### ISSUE TITLE
False-positive alternative Union syntax requires Python 3.10 with `asyncio.Future`

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#2204]
<h3> Environment data </h3>

-   Language Server version: 2021.12.2
-   OS and version: darwin x64
-   Python version: 3.9.9
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: strict

<h3> Expected behaviour </h3>

PEP 604 syntax can be used in type annotations if combined with `from __future__ import annotations`.

<h3> Actual behaviour </h3>

```
Alternative syntax for unions requires Python 3.10 or newer
```
On line 15 and 20

<img width="722" alt="Screen Shot 2022-01-01 at 18 26 24" src="https://user-images.githubusercontent.com/30130371/147856371-e34f6ebc-8051-4c33-a6df-e3a46707893a.png">

<h3> Logs </h3>
 --

<h3> Code Snippet / Additional information </h3>

```python
from __future__ import annotations

import asyncio
from typing import Any, overload

@overload
def async_run_job(
    target: int,
) -> asyncio.Future[int] | None:
    ...

@overload
def async_run_job(
    target: str,
) -> asyncio.Future[str] | None:
    ...

def async_run_job(
    target: int | str
) -> asyncio.Future[Any] | None:
    ...
```

I was able to replicate the issue with `asyncio.Task` as well. Almost all other types seem to work as expected though.

**Comment by erictraut** [comment]
Thanks for the bug report. This is due to a bug in pyright that has already been fixed. It will be included in the next release of pylance.

**Comment by judej** [comment]
Verified fixed in 2021.12.2
